client/verifier.go

```diff
@@ -46,17 +47,21 @@ func (c *logVerifier) VerifyRoot(trusted, newRoot *trillian.SignedLogRoot,
 	consistency [][]byte) error {
 
 	// Verify SignedLogRoot signature.
-	hash := tcrypto.HashLogRoot(*newRoot)
-	if err := tcrypto.Verify(c.pubKey, hash, newRoot.Signature); err != nil {
-		return err
+	if (newRoot == nil){
+		return fmt.Errorf("VerifyRoot() error: newRoot == nil.")
+	}else {
+		hash := tcrypto.HashLogRoot(*newRoot)
+		if err := tcrypto.Verify(c.pubKey, hash, newRoot.GetSignature()); err != nil {
+			return err
+		}
 	}
 
 	// Implicitly trust the first root we get.
-	if trusted.TreeSize != 0 {
+	if trusted.GetTreeSize() != 0 {
```

Comment:
I would agree.  Clients should explicitly create an initial trusted root to work with.
This can easily be a static object with TreeSize == 0 and RootHash = empty root